src/irmin-pack/irmin_pack.ml

```diff
@@ -399,6 +422,8 @@ struct
         Commit.CA.v ~fresh ~readonly ~lru_size ~index root >>= fun commit ->
         Branch.v ~fresh ~readonly root >|= fun branch ->
         { contents; node; commit; branch; config; index }
+
+      let close t = Contents.CA.close (contents_t t)
```

Comment:
for consistency, try to close all the stores and let the logic counting the number of shared fds do its magic.

Reply:
Agree, but couldn't close the stores before, as the close operation wasn't available in the API.

Reply:
fair enough :-) I'll merge #845 first so you could rebase your PR on top of it!